src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -51,5 +52,43 @@ public static void ThrowInvalidOperationException_SerializerPropertyNameNull(Jso
         {
             throw new InvalidOperationException(SR.Format(SR.SerializerPropertyNameNull, jsonClassInfo.Type.FullName, jsonPropertyInfo.PropertyInfo.Name));
         }
+
+        public static void ThrowJsonSerializationException_DeserializeDataRemaining(long length, long bytesRemaining)
+        {
+            throw new JsonSerializationException(SR.Format(SR.DeserializeDataRemaining, length, bytesRemaining), 0, 0, "");
+        }
+
+        public static void ThrowJsonSerializationException_DeserializeDuplicateKey(string key, in Utf8JsonReader reader, string path)
+        {
+            ThowJsonSerializationException(SR.Format(SR.DeserializeDuplicateKey, key), in reader, path);
+        }
+
+        private static void ThowJsonSerializationException(string message, in Utf8JsonReader reader, string path)
+        {
+            long lineNumber = reader.CurrentState._lineNumber;
```

Comment:
nit: Store `reader.CurrentState` in a local.

Reply:
The state shouldn't be in a local because it is a struct and a copy will be made.

Reply:
Why is a struct copy problematic?

Reply:
Not so bad here since no changes are made to the struct, but in other places mentioning the same "make a local" comment it changes the semantics\logic. I plan on keeping it as-is here to prevent any unnecessary copies.